File: Anurag_19_CSE/Java/day26.java
//Link For Problem: https://leetcode.com/problems/number-of-1-bits/

public class day26 {

    /*
     * TC : O(32) ==> O(1)
     * 
     * SC : O(1)
     */

    public int hammingWeight(int n) {
        int count = 0;

        while (n != 0) {
            count++;
            n = n & (n - 1);
        }

        return count;
    }
}

File: Anurag_19_CSE/Python/day26.py
# Link For Problem: https://leetcode.com/problems/number-of-1-bits/

class Solution:

    """
        TC : O(32) ==> O(1)

        SC : O(1)
    """

    def hammingWeight(self, n: int) -> int:
        ans: int = 0

        while n:
            ans += 1
            n = n & (n-1)

        return ans
